runners/maven-plugin/src/main/kotlin/DokkaMojo.kt

```diff
@@ -254,52 +250,27 @@ abstract class AbstractDokkaMojo(private val defaultDokkaPlugins: List<Dependenc
         gen.generate()
     }
 
-    private fun newRepositorySystem(): RepositorySystem {
-        val locator: DefaultServiceLocator = MavenRepositorySystemUtils.newServiceLocator()
-        locator.addService(RepositoryConnectorFactory::class.java, BasicRepositoryConnectorFactory::class.java)
-        locator.addService(TransporterFactory::class.java, FileTransporterFactory::class.java)
-        locator.addService(TransporterFactory::class.java, HttpTransporterFactory::class.java)
-        return locator.getService(RepositorySystem::class.java)
-    }
-
-    private fun newSession(system: RepositorySystem): RepositorySystemSession {
-        val session: DefaultRepositorySystemSession =
-            MavenRepositorySystemUtils.newSession()
-        val localRepo = LocalRepository(System.getProperty("user.home") + "/.m2/repository")
-        session.localRepositoryManager = system.newLocalRepositoryManager(session, localRepo)
-        return session
-    }
-
     private fun getArtifactByAether(
         groupId: String,
         artifactId: String,
         version: String
     ): List<File> {
-        val repoSystem: RepositorySystem = newRepositorySystem()
-        val session: RepositorySystemSession = newSession(repoSystem)
-        val dependency =
-            org.eclipse.aether.graph.Dependency(DefaultArtifact("$groupId:$artifactId:$version"), "compile")
-        val collectRequest = CollectRequest()
-        collectRequest.root = dependency
-        val repositories: List<RemoteRepository> =
-            (mavenProject?.remoteProjectRepositories?.plus(mavenProject?.remotePluginRepositories ?: emptyList())
-                ?: mavenProject?.remotePluginRepositories ?: emptyList())
-        repositories.forEach {
-            collectRequest.addRepository(
-                RemoteRepository.Builder(
-                    "repo",
-                    "default",
-                    it.url
-                ).setAuthentication( it.authentication ).build()
-            )
-        }
-        val node: DependencyNode = repoSystem.collectDependencies(session, collectRequest).root
-        val dependencyRequest = DependencyRequest()
-        dependencyRequest.root = node
-        repoSystem.resolveDependencies(session, dependencyRequest)
-        val nlg = PreorderNodeListGenerator()
-        node.accept(nlg)
-        return nlg.files
+
+        val request: ArtifactResolutionRequest = ArtifactResolutionRequest()
+        request.isResolveRoot = true
+        request.isResolveTransitively = true
+        request.localRepository = session!!.localRepository
+        request.remoteRepositories = mavenProject!!.pluginArtifactRepositories
```

Comment:
Only the configured plugin repositories are considered. This could be easily changed to consider also the project repositories:

```
request.remoteRepositories = mavenProject!!.remoteArtifactRepositories.plus(mavenProject!!.pluginArtifactRepositories)
```

Reply:
That should not be necessary IMHO